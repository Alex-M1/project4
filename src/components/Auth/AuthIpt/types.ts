import { TCredentialsAction } from 'store/credentials/types';
import { Dispatch } from 'react';

export interface IAuthIndex {
  type: 'login' | 'password' | 'confirm';
  page?: 'auth' | 'registration';
}

export type TDispatch = Dispatch<TCredentialsAction>

export interface IAuthIpt extends IAuthIndex {
  value: string;
  placeholder: string;
  onChange: (value: string) => void;
}
