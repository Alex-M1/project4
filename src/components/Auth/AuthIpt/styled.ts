import styled from 'styled-components';

export const StInputDiv = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    padding: 25px 0;
`