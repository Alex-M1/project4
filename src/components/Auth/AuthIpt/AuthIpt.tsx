import React from 'react';
import { useTranslation } from 'react-i18next';

import { useTheme } from 'src/components/hooks/useTheme';
import Input from '../../_common_/Input';
import { IAuthIpt } from './types';
import { StInputDiv, StSpan } from './styled';

const AuthIpt = ({ type, value, onChange }: IAuthIpt) => {
  const { t } = useTranslation();
  const { colors, theme } = useTheme();
  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => onChange(e.target.value);
  return (
    <StInputDiv>
      <StSpan
        theme={theme}
        colors={colors}
      >
        {t(type)}
      </StSpan>
      <Input
        type={type}
        value={value}
        onChange={onInputChange}
      />
    </StInputDiv>
  );
};

export default AuthIpt;
