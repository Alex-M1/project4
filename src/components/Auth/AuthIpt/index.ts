import { connect } from 'react-redux';

import { AppStateType } from 'store/rootReducer';
import { getInputValue } from 'store/user/selectors';
import { setCredentialsValue } from 'store/user/action';
import AuthIpt from './AuthIpt';

export interface IAuthIndex {
  type: 'login' | 'password' | 'confirm';
  page?: 'auth' | 'registration';
}

const mapStateToProps = (state: AppStateType, props: IAuthIndex) => ({
  value: getInputValue(state, { page: 'auth', field: props.type }),
});

const mapDispatchToProps = {
  setCredentialsValue,
};
export default connect(mapStateToProps, mapDispatchToProps)(AuthIpt);
