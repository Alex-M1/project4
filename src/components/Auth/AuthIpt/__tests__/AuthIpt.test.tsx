import React from 'react';
import { mount, shallow } from 'enzyme';
import AuthIpt from '../AuthIpt';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
        t: jest.fn(),
    })),
}),
);

describe('AuthIpt', () => {
    let props;
    beforeEach(() => {
        props = {
            type: 'login',
            value: 'value',
            setCredentialsValue: jest.fn(),
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<AuthIpt {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StInputDiv', () => {
        const component = mount(<AuthIpt {...props} />);
        expect(component.find('styled__StInputDiv')).toHaveLength(1);
    });
    it('should render StSpan', () => {
        const component = mount(<AuthIpt {...props} />);
        expect(component.find('styled__StSpan')).toHaveLength(1);
    });
    it('should call setCredentialsValue', () => {
        const component = mount(<AuthIpt {...props} />);
        component.find('styled__StInput').simulate('change', { target: { value: 'testValue' } });
        expect(component.props().setCredentialsValue).toHaveBeenCalled();
    });
    it('should change setCredentialsValue', () => {
        const component = mount(<AuthIpt {...props} />);
        const ipt = component.find('Input');
        ipt.props().onChange({ target: { value: 'sdads' } });
        expect(component.props().setCredentialsValue).toHaveBeenCalled();
    });
});
