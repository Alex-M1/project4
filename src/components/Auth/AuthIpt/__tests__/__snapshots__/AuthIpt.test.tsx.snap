// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthIpt Should match snapshot 1`] = `"<div class=\\"styled__StInputDiv-iTvJvh gqdFEH\\"><span class=\\"styled__StSpan-bbjaPg hFPkpt\\"></span><input type=\\"login\\" value=\\"value\\" class=\\"styled__StInput-ifVcmW fkrHSd\\"/></div>"`;
